### PR TITLE
Update readme for CBMC local runs

### PR DESCRIPTION
**Pull request by kstribrnAmzn**
Description
-----------
* Adds readme section describing CBMC local execution
* Corrects mispelling of 'pieces'
* Adds comment explaining thingname size bounds

Checklist:
----------

- [N/A] I have tested my changes. No regression in existing tests.
- [N/A] My code is formatted using Uncrustify.
- [ x ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,23 @@ The `test/cbmc/proofs` directory contains CBMC proofs.
...
 
+### CBMC Locally
+To run a single CBMC proof locally, you can build the Makefile in any of the CBMC proofs. The Makefile is located in the `test/cbmc/proof/<the proof you want>/` directory.
+
+Running `make` will produce a HTML-based report nearly identical to the one produced by the CI step.
+
+**A couple notes about CBMC Proofs**
+* Mac doesn't implement POSIX message queues (mqueue.h)
+* It is possible that Mac fails to recognize your loop unwinding identifiers. For this case, you'll want to use the `__builtin___<function>_chk` identifier.
+  * For example, the **requestJob_Mqtt** proof  fails on my MAC with the following error
```
Comment (feliperodri) [inline README.md:99]:
```suggestion
  * For example, the **requestJob_Mqtt** proof fails on macOS with the following error:
```

File: README.md [hunk 2]
```diff
@@ -88,6 +88,23 @@ The `test/cbmc/proofs` directory contains CBMC proofs.
...
+
+```
+Loop unwinding failures
+[trace] __builtin___strncpy_chk.unwind.0 in line 36 in file <builtin-library-__builtin___strncpy_chk>
+```
+
+To solve this, replace `strncpy` with `__builtin___strncpy_ch on [this line](https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/main/test/cbmc/proofs/requestJob_Mqtt/Makefile#L16).
```
Comment (feliperodri) [inline README.md:106]:
```suggestion
To solve this issue, replace `strncpy` with `__builtin___strncpy_ch` on [this line](https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/main/test/cbmc/proofs/requestJob_Mqtt/Makefile#L16).
```

**Review by feliperodri — CHANGES_REQUESTED** [review]

**Comment by kstribrnAmzn** [comment]
Thanks for the review and suggestions @feliperodri! I've applied them.

**Review by feliperodri — APPROVED** [review]
LGTM 🚢 it!

**Pull request merged**